doc/design/cluster_train/trainer.md

```diff
@@ -0,0 +1,82 @@
+# Design Doc: Trainer Communication Library
+
+For an overview of trainer's role, please refer to [distributed training design doc](README.md). In this design doc, we will discuss the trainer's communication library, which will manage communication with parameter servers and the [master server](master_server.md). The library will be implemented in [Go](https://golang.org/) and made available as a static or dynamic library with a C header file.
+
+## Go Interface
+
+The Go interface is the basic abstraction of communications with the master server and parameter servers. We will add another layer on top (add retry logic, polish interface with C idiom) before exposing the library with a [C interface](#c-interface).
+
+```go
```

Comment:
Are these interfaces kind of duplicate to https://github.com/PaddlePaddle/Paddle/pull/1964? The communication interfaces are export by the server and will be called remotely by the client. So the server and the client need to use the same piece of interface definition code.